src/socket-message-handling.ts

```diff
@@ -41,14 +41,32 @@ const setupWSConnection = (conn) => {
    */
   const subscribedTopics = new Set();
   let closed = false;
+
   // Check if connection is still alive
-  let pongReceived = true;
-  const pingInterval = setInterval(() => {
-    if (!pongReceived) {
+  let providerInterfacePongReceived = true;
+  const providerInterfacePingInterval = setInterval(() => {
+    if (!providerInterfacePongReceived) {
+      conn.close();
+      clearInterval(providerInterfacePingInterval);
+    }
+    {
+      providerInterfacePongReceived = false;
+      try {
+        conn.ping();
+      } catch (e) {
+        conn.close();
+        clearInterval(providerInterfacePingInterval);
+      }
+    }
+  }, pingTimeout);
```

Comment:
I don't understand why we're adding this. This is the same as the other interval.